### PR TITLE
Testing Your Ruby Code - Enumerable

### PR DESCRIPTION
**Pull request by mertcihankurel**
### Rspec testing for Enumerable Methods
**İn this project:**
1-) We added all test cases.
2-) We covered all edge cases, empty inputs. Block, procs, argument, and symbols.
3-) Fixed all rubocop errors.

### CODE REVIEW

**Review by Dandush03 — CHANGES_REQUESTED** [review]
# Status: :red_circle: Require Changes :red_circle: 
### You're almost done! :partying_face: 

- [ ] #multiply_els not been tested 
### Since we are learning TDD we need to be sure to test each scenario possible in our code.
# Happy Coding :computer: :nerd_face:

File: main.rb [hunk 1]
```diff
@@ -1,145 +1,174 @@
...
     end
     result
   end
+
+  def multiply_els(array)
+    raise ArgumentError('Only arrays with Numeric elements accepted') unless array.my_all?(Numeric)
+
+    array.my_inject(:*)
+  end
 end
```
Comment (kobiyoyo) [inline main.rb]: You have a duplicate of this `multiply_els(array)` ,lets remove one

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [ ] Some test are failing, lets fix that.
- [ ] Lets remove this in the main.rb .
````ruby
array = [5, 9, 10]
my_proc = proc { |num| num > 6 }
p [2, 54, 6, 7].my_map(my_proc) { |num| num < 10 }
p 5.times.my_inject(20, :*)
p 5.times.inject(20, :*)
p multiply_els(array)
````
- [ ] check the comment below

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [ ] Some tests are still failing, let's fix that.



- [ ] Lets remove this code that is in the main.rb .
````ruby
array = [5, 9, 10]
my_proc = proc { |num| num > 6 }
p [2, 54, 6, 7].my_map(my_proc) { |num| num < 10 }
p 5.times.my_inject(20, :*)
p 5.times.inject(20, :*)
p multiply_els(array)
````
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
# Project Approved :thumbsup:
## Hello Team  ,  you have done a brilliant job with this project. :clap: 
Just a few minor issues to fix
- [ ] lets rename the file from `enumerables_testing.rb` to `enumerables_testing_spec.rb`
- [ ] In the `enumerables_testing_spec.rb` change the main.rb file path to `require_relative '../main.rb'`
#### You can merge once done  !!!

**Review by kobiyoyo — APPROVED** [review]
Project Approved
